pandasai/helpers/anonymizer.py

```diff
@@ -175,5 +178,12 @@ def anonymize_dataframe_head(
                 pd.eval(cell_value) if cell_value in ["True", "False"] else cell_value
             )
     # restore the original data types
-    data_frame = data_frame.astype(dtypes)
+    for i in range(len(data_frame.columns)):
+        dt = data_frame.dtypes[i]
+        col = data_frame.columns[i]
+        if dt == "Int64":
+            data_frame[col] = data_frame[col].astype("float64")
```

Comment:
Why do you convert to `float64` to then convert to `int64`?

Reply:
actually, as in the code, we are converting the data frame to string, but when the data frame is Int64 type and in the anonymizer algorithm converting the string back to its original Int64 type is not doable. To convert the string back to Int64 type will need to first convert it to float64 and then Int64. (bec. float64 handles null values(np.nan) as well).

Reply:
Great point, I missed it! Thanks a lot for clarifying!

Reply:
I am facing the same issue, and I don't know even after using the above solution, I still get the error, Int64 not understood

Reply:
![image](https://github.com/gventuri/pandas-ai/assets/80671468/3f9c9602-7496-4439-a27c-a17728fbd0c0)